src/service-override/views.ts

```diff
@@ -304,16 +334,45 @@ function registerEditorPane (options: EditorPanelOption): { disposable: IDisposa
     }
   }
 
-  const disposable = Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
+  const disposableStore = new DisposableStore()
+
+  disposableStore.add(Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
     EditorPaneDescriptor.create(
       CustomEditorPane,
       options.id,
       options.name
     ),
-    [new SyncDescriptor(CustomEditorInput)])
+    [new SyncDescriptor(CustomEditorInput)]))
+
+  const registerEditorOptions = options.registerEditorOptions
+  if (registerEditorOptions != null) {
+    getService(IEditorResolverService).then(editorResolverService => {
+      if (disposableStore.isDisposed) {
+        return
+      }
+
+      disposableStore.add(editorResolverService.registerEditor(
+        registerEditorOptions.globPattern,
+        {
+          id: CustomEditorInput.ID,
+          label: options.name,
+          priority: registerEditorOptions.priority ?? RegisteredEditorPriority.default
+        },
+        {},
+        {
+          createEditorInput (editorInput: IResourceEditorInput | ITextResourceEditorInput, _group: IEditorGroup) {
+            return {
+              options: {},
+              editor: new CustomEditorInput(undefined, editorInput)
```

Comment:
That's a shame we can't provide a `IEditorCloseHandler` in that case :)

Reply:
Actually... Thinking about it, we can enable that, but then it has to be passed to `registerEditorPane`. I can make that change if you want.

Reply:
or passed to the potentially new `registerEditor` returned function (:

Reply:
Will add thise!

Reply:
Done!